src/sync/persist-fixup-transformer.ts

```diff
@@ -0,0 +1,68 @@
+import type * as kv from '../kv/mod';
+import * as dag from '../dag/mod';
+import type {HashType} from '../db/hash-type';
+import * as db from '../db/mod';
+import type {Hash} from '../hash';
+import {assert} from '../asserts';
+
+type OldHash = Hash;
+type NewHash = Hash;
+type Mappings = ReadonlyMap<OldHash, NewHash>;
+
+/**
+ * This transformer is used to fixup the hashes we got from the source dag (the
+ * perdag in our case) and rewrite the required chunks in the destination dag
+ * (the memdag).
+ */
+export class PersistFixupTransformer extends db.Transformer {
+  private readonly _mappings: Mappings;
+
+  constructor(dagWrite: dag.Write, mappings: Mappings) {
+    super(dagWrite);
+    this._mappings = mappings;
+  }
+
+  override async transformCommit(
```

Comment:
TODO(arv) Maybe this could be done as `shouldSkipChunk`? I feel like all these transformers do  something like that